python/ml4ir/base/model/scoring/prediction_helper.py

```diff
@@ -86,10 +89,23 @@ def _filter_records(x, mask):
     @tf.function
     def _predict_score(features, label):
         """Predict scores and compute additional output from input features using the input model"""
+        # inference with Monte Carlo
         if is_compiled:
+            # inference with no dropout
             scores = infer(features)[output_name]
+            # running monte carlo iterations
+            for _ in range(monte_carlo_inference_trials):
+                s = infer(features, training=True)[output_name]
+                scores += s
+            scores /= (monte_carlo_inference_trials+1)
         else:
+            # inference with no dropout
             scores = infer(**features)[output_name]
+            # running monte carlo iterations
+            for _ in range(monte_carlo_inference_trials):
+                s = infer(**features, training=True)[output_name]
+                scores += s
+            scores /= (monte_carlo_inference_trials+1)
```

Comment:
@mohazahran I am not sure why this is needed. The way it is implemented the MC trials are already part of the `infer`. Now you are doing a double run as each `infer` will run for `monte_carlo_inference_trials`